packages/unchained-client/src/evm/parser/erc20.ts

```diff
@@ -58,9 +57,9 @@ export class Parser<T extends Tx> implements SubParser<T> {
 
     switch (txSigHash) {
       case this.supportedFunctions.approveSigHash: {
-        const amount = decoded.args.amount as BigNumber
+        const amount = decoded.args.amount as BigInt
         const value = amount.toString()
-        if (amount.isZero()) {
+        if (amount === BigInt(0)) {
```

Comment:
non-blocking, and not sure of your preference, but `0n === BigInt(0)`

```suggestion
        if (amount === 0n {
```

Reply:
Might as well - 48323359c2